src/eca/features/chat.clj

```diff
@@ -42,12 +42,237 @@
     (db/update-workspaces-cache! @db*)))
 
 (defn ^:private assert-chat-not-stopped! [{:keys [chat-id db*] :as chat-ctx}]
-  (when (identical? :stoping (get-in @db* [:chats chat-id :status]))
+  (when (identical? :stopping (get-in @db* [:chats chat-id :status]))
     (finish-chat-prompt! :idle chat-ctx)
     (logger/info logger-tag "Chat prompt stopped:" chat-id)
     (throw (ex-info "Chat prompt stopped" {:silent? true
                                            :chat-id chat-id}))))
 
+;;; Helper functions for tool call state management
```

Comment:
do we need this kind of line comment?

Reply:
We don't **need** it, but I find it useful to separate the helpers from the main code.
I can remove if you prefer.

Reply:
yeah, I wonder if that doesn't show how this bunch of logic should not belong to this ns maybe, but seems to overcomplicate it

Reply:
Yeah, I meant the whole PR is refactoring this logic and has lots of code related to it, but I understand it's a feature that is on a complex stage (complex meaning needs a state machine).
It's ok for now, let's see how this evolves in the future